### DISCUSSION TITLE
When I set the atom value(with storage), 'atom.write is not a  function' error is thrown.

### DISCUSSION THREAD

**Discussion by engadori** [pmndrs/jotai#1824] [Q&A]
Hello jotai.
I run into difficulties when I set the new value to my atom..

I have an 'atomWithStorage' object that's initial value is referenced from another atom
`export const fooAtom = atomWithStorage('storage-key-1', (get:Getter) => get(someAtom), sessionStorage)` 

When I set the new value with useSetAtom method, the error says 'Uncaught TypeError: atom.write is not a function'.

I just setting the initial value, but wonder why the atom setting method is removed..
And how can i fix it? my jotai version is 1.13.1

**Comment by dai-shi** [comment]
Unfortunately, the initial value has to be a value, not a function.
`atomWithDefault` util is something, but may not fit your needs.

**Reply by wootra** [reply]
in my knowledge, when you set getter or setter, it is considered as derived atom. In above case, it is not very clear adding function in your sessionStorage. ( sessionStorage will not even get function reference since it is not serializable ). I think you should think about your use case once more.

**Reply by wootra** [reply]
one more thing... once you use 1.13.X version, you might need to use useAtom, useAtomValue, useSetAtom from 'jotai/react', not from 'jotai'.
